File: 2261-k-divisible-elements-subarrays/2261-k-divisible-elements-subarrays.py
class Solution:
    def countDistinct(self, nums: List[int], k: int, p: int) -> int:
        res = set()
        n = len(nums)
        for i in range(n):
            l, ct = [], 0
            for j in range(i, n):
                if not nums[j] % p:
                    ct += 1
                l.append(nums[j])
                if ct <= k:
                    res.add(tuple(l))
        return len(res)

File: 2261-k-divisible-elements-subarrays/2261-k-divisible-elements-subarrays.cpp
class Solution {
public:
    int countDistinct(vector<int>& nums, int k, int p) {
        set<vector<int>> s;
        int n = nums.size();
        for(int i = 0; i < n; ++i) {
            vector<int> v;
            int ct = 0;
            for(int j = i; j < n; ++j) {
                if (nums[j]%p == 0) {
                    ++ct;
                }
                v.push_back(nums[j]);
                if (ct <= k) {
                    s.insert(v);
                } else {
                    break;
                }
            }
        }
        return s.size();
    }
};